apps/backend/src/libs/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,55 @@
+import { type FastifyReply, type FastifyRequest } from "fastify";
+import fp from "fastify-plugin";
+
+import { ExceptionMessage } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+import { token } from "~/libs/modules/token/token.js";
+import { UserError, userService } from "~/modules/users/users.js";
+
+import { WHITE_ROUTES } from "../libs/enums/enums.js";
+
+const authorization = fp((fastify, _, done) => {
+	fastify.decorateRequest("user", null);
+
+	fastify.addHook(
+		"preHandler",
+		async (request: FastifyRequest, reply: FastifyReply) => {
+			const { url } = request.raw;
+
+			for (const whiteRoute of WHITE_ROUTES) {
+				if (whiteRoute.test(url as string)) {
+					return;
+				}
+			}
+
+			const authToken = request.headers["authorization"];
+
+			if (!authToken) {
+				return await reply
+					.code(HTTPCode.UNAUTHORIZED)
+					.send({ error: ExceptionMessage.NO_TOKEN_PROVIDED });
+			}
+
+			try {
+				const { userId } = token.decodeToken(authToken);
```

Comment:
```suggestion
				const { payload } = await token.verify(jwtToken);
				const { userId } = payload;
```
you need to verify it instead, to prevent invalid/expired tokens